schema/validator.go

```diff
@@ -92,7 +99,56 @@ func (v unimplemented) Validate(src io.Reader) error {
 	return fmt.Errorf("%s: unimplemented", v)
 }
 
-func validateManifestDescendants(r io.Reader) error {
+type validateRefMediaFunc func(r io.Reader) error
+
+var vlidateRefMediaMap = map[Validator]validateRefMediaFunc{
+	MediaTypeManifestList: validateManifestListRefObject,
+	MediaTypeManifest:     validateManifestRefObject,
+}
+
+// ValidateRefMedia validates the referenced objects against OCI media type defined in spec.
+// r: the given reader.
+// v: the expected media type.
+func ValidateRefMedia(r io.Reader, v Validator) error {
+	f, ok := vlidateRefMediaMap[v]
+	if ok {
+		if f == nil {
+			return fmt.Errorf("referenced media type validation %q unimplemented", v)
+		}
+
+		buf, err := ioutil.ReadAll(r)
+		if err != nil {
+			return errors.Wrap(err, "unable to read the document file")
+		}
+
+		return f(bytes.NewReader(buf))
+	}
+
+	return nil
+}
+
+func validateManifestListRefObject(r io.Reader) error {
+	header := v1.ManifestList{}
+
+	buf, err := ioutil.ReadAll(r)
+	if err != nil {
+		return errors.Wrapf(err, "error reading the io stream")
+	}
+
+	err = json.Unmarshal(buf, &header)
+	if err != nil {
+		return errors.Wrap(err, "manifest list format mismatch")
+	}
+
+	for _, manifest := range header.Manifests {
+		if manifest.MediaType != string(v1.MediaTypeImageManifest) {
+			return fmt.Errorf("manifest %s has an unknown media type: %s", manifest.Digest, manifest.MediaType)
```

Comment:
@wking 
After former proposal from @stevvooe, I think we have made agreement. Current commit follows up the proposal thought, to use function slice. Consumer can select objects(fields, media type...) at will.
> I don't see a way to invoke this to error out when a media type is truly invalid (i.e. not compliant with RFC 6838, #448) vs. not necessarily supported by an OCI-compliant implementation (e.g. a non-OCI type). 

Under truly invalid case, JSON schema prompts error.
Under non-OCI type, validator will prompt error only if ref-type(here is media type) function is selected.
Honest I'm not very willing roll back the former strict framework, which has been rejected. And we had better to wait @stevvooe's feedback.
cc @stevvooe WDYT?